sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/text/TextTableProviderTest.java

```diff
@@ -17,69 +17,229 @@
 
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  @Rule
+  public TemporaryFolder tempFolder =
+      new TemporaryFolder() {
+        @Override
+        protected void after() {}
+      };
+
+  private static final String SQL_CSV_SCHEMA = "(f_string VARCHAR, f_int INT)";
+  private static final Schema CSV_SCHEMA =
+      Schema.builder()
+          .addNullableField("f_string", Schema.FieldType.STRING)
+          .addNullableField("f_int", Schema.FieldType.INT32)
+          .build();
+
+  private static final Schema LINES_SCHEMA = Schema.builder().addStringField("f_string").build();
+  private static final String SQL_LINES_SCHEMA = "(f_string VARCHAR)";
+
+  // Even though these have the same schema as LINES_SCHEMA, that is accidental; they exist for a
+  // different purpose, to test Excel CSV format that does not ignore empty lines
+  private static final Schema SINGLE_STRING_CSV_SCHEMA =
+      Schema.builder().addStringField("f_string").build();
+  private static final String SINGLE_STRING_SQL_SCHEMA = "(f_string VARCHAR)";
+
+  /**
+   * Tests {@code CREATE TABLE TYPE text} with no format reads a default CSV.
+   *
+   * <p>The default format ignores empty lines, so that is an important part of this test.
+   */
   @Test
-  public void testGetTableType() throws Exception {
-    assertEquals("text", provider.getTableType());
+  public void testLegacyDefaultCsv() throws Exception {
+    Files.write(
+        tempFolder.newFile("test.csv").toPath(),
+        "hello,13\n\ngoodbye,42\n".getBytes(Charsets.UTF_8));
+
+    BeamSqlEnv env = BeamSqlEnv.inMemory(new TextTableProvider());
+    env.executeDdl(
+        String.format(
+            "CREATE TABLE test %s TYPE text LOCATION '%s/*'",
+            SQL_CSV_SCHEMA, tempFolder.getRoot()));
+
+    PCollection<Row> rows =
+        BeamSqlRelUtils.toPCollection(pipeline, env.parseQuery("SELECT * FROM test"));
+
+    PAssert.that(rows)
+        .containsInAnyOrder(
+            Row.withSchema(CSV_SCHEMA).addValues("hello", 13).build(),
+            Row.withSchema(CSV_SCHEMA).addValues("goodbye", 42).build());
+    pipeline.run();
   }
 
+  /**
+   * Tests {@code CREATE TABLE TYPE text} with a format other than "csv" or "lines" results in a CSV
+   * read of that format.
+   */
   @Test
-  public void testBuildBeamSqlTable() throws Exception {
-    Table table = mockTable("hello", null);
-    BeamSqlTable sqlTable = provider.buildBeamSqlTable(table);
+  public void testLegacyTdfCsv() throws Exception {
+    Files.write(
+        tempFolder.newFile("test.csv").toPath(),
+        "hello\t13\n\ngoodbye\t42\n".getBytes(Charsets.UTF_8));
 
-    assertNotNull(sqlTable);
-    assertTrue(sqlTable instanceof BeamTextCSVTable);
+    BeamSqlEnv env = BeamSqlEnv.inMemory(new TextTableProvider());
+    env.executeDdl(
+        String.format(
+            "CREATE TABLE test %s TYPE text LOCATION '%s/*' TBLPROPERTIES '{\"format\":\"TDF\"}'",
```

Comment:
@XuMingmin  This is a test of TSV actually. We could register `TSV` as an alias for `TDF` once we move to our own enum.